community/kernel/src/main/java/org/neo4j/kernel/builtinprocs/IndexSpecifier.java

```diff
@@ -82,16 +92,16 @@ private String either( String first, String second )
 
     private static String or( String first, String second )
     {
-        return "(:?" + first + "|" + second + ")";
+        return "(?:" + first + "|" + second + ")";
     }
 
-    private static String simpleIdentifier( String name )
+    private static String identifier()
     {
-        return format( "(?<%s>[A-Za-z0-9_]+)", name );
+        return "([A-Za-z0-9_]+)";
     }
 
-    private static String complexIdentifier( String name )
+    private static String qoutedIdentifier()
     {
-        return format( "(?:`(?<%s>[^`]+)`)", name );
+        return "(?:`([^`]+)`)";
```

Comment:
Quoted identifier creates a non-capturing group, but the identifier creates a capturing group…

I _suspect_ that more tests can be written for this.